store/iavl/tree.go

```diff
@@ -23,20 +23,19 @@ type (
 		Set(key, value []byte) (bool, error)
 		Remove(key []byte) ([]byte, bool, error)
 		SaveVersion() ([]byte, int64, error)
-		DeleteVersion(version int64) error
-		DeleteVersions(versions ...int64) error
 		Version() int64
 		Hash() ([]byte, error)
 		WorkingHash() ([]byte, error)
 		VersionExists(version int64) bool
+		DeleteVersionsTo(version int64) error
 		GetVersioned(key []byte, version int64) ([]byte, error)
 		GetImmutable(version int64) (*iavl.ImmutableTree, error)
 		SetInitialVersion(version uint64)
 		Iterator(start, end []byte, ascending bool) (types.Iterator, error)
 		AvailableVersions() []int
-		LoadVersionForOverwriting(targetVersion int64) (int64, error)
-		LazyLoadVersionForOverwriting(targetVersion int64) (int64, error)
+		LoadVersionForOverwriting(targetVersion int64) error
 		TraverseStateChanges(startVersion, endVersion int64, fn func(version int64, changeSet *iavl.ChangeSet) error) error
+		SaveChangeSet(changeSet *iavl.ChangeSet) (int64, error)
```

Comment:
do we still need this api change?

Reply:
it can be useful when create the migrating tool of the archived node.

Reply:
This still needs to be implemented.